### PR TITLE
Add footnotes to docstring of numericallyapunov

### PR DESCRIPTION
**Pull request by asinghvi17**

### CODE REVIEW

**Comment by Datseris** [comment]
? :P

**Comment by asinghvi17** [comment]
The MD footnote syntax is `[^n]`, so these would previously not link to the footnote.  Hopefully, they will now...

**Comment by Datseris** [comment]
Well, I will have to ask you to check this by building the docs before we can merge this... I understand it is convoluted, because you need master of DynamicalSystems.jl and ChaosTools.jl, but I think it is worth it to check. If this works, I will also do it everywhere I add references. 

(p.s.: if it works increase patch version of ChaosTools)

**Comment by asinghvi17** [comment]
I think I found a hacky solution to test this which involves no local effort 😄.  Let's see if it works...

**Comment by asinghvi17** [comment]
@Datseris I can confirm this works:
https://juliadynamics.github.io/DynamicalSystems.jl/previews/PR109/chaos/nlts/#ChaosTools.numericallyapunov

**Comment by asinghvi17** [comment]
@Datseris I'm essentially done working on this branch; we could merge it now, but feel free to push to it if you want to get every docstring to have footnotes.

**Comment by Datseris** [comment]
yes please let it open, I'll try to modify as many as possible. And also remove all `## References` headers.

**Comment by Datseris** [comment]
@asinghvi17 can you please review and merge, to ensure I've used correct syntax?

File: src/chaosdetection/partially_predictable.jl [hunk 1]
```diff
@@ -6,11 +6,11 @@ export predictability
     predictability(ds::DynamicalSystem; kwargs...) -> chaos_type, ν, C
 
 Determine whether `ds` displays strongly chaotic, partially-predictable chaotic
-or regular behaviour, using the method by Wernecke et al. described in [1].
+or regular behaviour, using the method by Wernecke et al. described in[1].
```
Comment (asinghvi17) [inline src/chaosdetection/partially_predictable.jl]: Needs update

File: src/orbitdiagrams/poincare.jl [hunk 2]
```diff
@@ -59,14 +61,7 @@ planecrossing = PlaneCrossing(plane, direction > 0)
 ```
 and `idxs` must be `Int` or `SVector{Int}`.
 
-## References
-[1] : H. Poincaré, *Les Methods Nouvelles de la Mécanique Celeste*,
```
Comment (asinghvi17) [inline src/orbitdiagrams/poincare.jl]: Had this been re-added (does it need to be)?

File: src/periodicity/period.jl [hunk 3]
```diff
@@ -56,13 +56,13 @@ These methods tend to be slow, but versatile and low-error.
   It deffaults to `mean(v)`.
 
 For more information on the periodogram methods, see the documentation of
-`DSP.jl` and `LombScargle.jl`.
+[DSP.jl and LombScargle.jl.
```
Comment (asinghvi17) [inline src/periodicity/period.jl]: Extra [ in front

File: src/chaosdetection/partially_predictable.jl (hunk 1 above)
Comment (Datseris) [reply src/chaosdetection/partially_predictable.jl]:
```suggestion
or regular behaviour, using the method by Wernecke et al. described in[^Wernecke2017].
```

File: src/periodicity/period.jl (hunk 3 above)
Comment (Datseris) [reply src/periodicity/period.jl]:
```suggestion
DSP.jl and LombScargle.jl.
```

File: src/orbitdiagrams/poincare.jl (hunk 2 above)
Comment (Datseris) [reply src/orbitdiagrams/poincare.jl]: not needed

**Pull request merged**